src/main/java/org/opensrp/web/rest/PlanResource.java

```diff
@@ -372,5 +401,30 @@ public boolean isReturnCount() {
 		}
 
 	}
+
+	private PlanSearchBean createPlanSearchBean(boolean isTemplateParam, Integer pageNumber, Integer pageSize, String orderByType,
+			String orderByFieldName, String planStatus, Map<String, String> useContextFilters) {
+		PlanSearchBean planSearchBean = new PlanSearchBean();
+		planSearchBean.setExperimental(isTemplateParam);
+		if(pageNumber != null) {
+			planSearchBean.setPageNumber(pageNumber);
+		}
+		if(pageSize != null) {
+			planSearchBean.setPageSize(pageSize);
+		}
```

Comment:
are the if statements required here

Reply:
Setting fields in PlanSearchBean only when it's not null.

Reply:
Removed if statements to optimize complexity.